tests/integration_tests/functional/test_balloon.py

```diff
@@ -0,0 +1,713 @@
+        keyword="vmlinux-4.14"
+    ))
+    disk_artifacts = ArtifactSet(artifacts.disks(keyword="ubuntu"))
+
+    # Create a test context and add builder, logger, network.
+    test_context = TestContext()
+    test_context.custom = {
+        'builder': MicrovmBuilder(bin_cloner_path),
+        'network_config': network_config,
+        'logger': logger,
+        'snapshot_type': SnapshotType.FULL,
+        'seq_len': 5
+    }
+
+    # Create the test matrix.
+    return TestMatrix(
+        context=test_context,
+        artifact_sets=[
+            microvm_artifacts,
+            kernel_artifacts,
+            disk_artifacts
+        ]
+    )
+
+
+def create_vm(context, deflate=True, stats_interval=0):
+    """Create a microvm with a balloon device."""
+    logger = context.custom['logger']
+    vm_builder = context.custom['builder']
+    snapshot_type = context.custom['snapshot_type']
+    enable_diff_snapshots = snapshot_type == SnapshotType.DIFF
+
+    logger.info(
+        "Testing {} with microvm: \"{}\", kernel {}, disk {} "
+        .format(
+            snapshot_type,
+            context.microvm.name(),
+            context.kernel.name(),
+            context.disk.name()
+        )
+    )
+
+    # Create a rw copy artifact.
+    root_disk = context.disk.copy()
+    # Get ssh key from read-only artifact.
+    ssh_key = context.disk.ssh_key()
+    # Create a fresh microvm from aftifacts.
+    microvm = vm_builder.build(
+        kernel=context.kernel,
+        disks=[root_disk],
+        ssh_key=ssh_key,
+        config=context.microvm,
+        enable_diff_snapshots=enable_diff_snapshots
+    )
+
+    network_config = net_tools.UniqueIPv4Generator.instance()
+    _, host_ip, guest_ip = microvm.ssh_network_config(
+        network_config,
+        '1',
+        tapname="tap0"
+    )
+    logger.debug("Host IP: {}, Guest IP: {}".format(host_ip, guest_ip))
+
+    copy_fillmem_to_rootfs(root_disk.local_path())
+
+    # Add a memory balloon with stats enabled.
+    response = microvm.balloon.put(
+        amount_mb=0,
+        deflate_on_oom=deflate,
+        must_tell_host=False,
+        stats_polling_interval_s=stats_interval
+    )
+    assert microvm.api_session.is_status_no_content(response.status_code)
+
+    return microvm
+
+
+def copy_fillmem_to_rootfs(rootfs_path):
+    """Build and copy the 'memfill' program to the rootfs."""
+    subprocess.check_call("gcc ./host_tools/fillmem.c -o fillmem", shell=True)
+    subprocess.check_call("mkdir tmpfs", shell=True)
+    subprocess.check_call("mount {} tmpfs".format(rootfs_path), shell=True)
+    subprocess.check_call("cp fillmem tmpfs/sbin/fillmem", shell=True)
+    subprocess.check_call("rm fillmem", shell=True)
+    subprocess.check_call("umount tmpfs", shell=True)
+    subprocess.check_call("rmdir tmpfs", shell=True)
+
+
+# pylint: disable=C0103
+@pytest.mark.skipif(
+    platform.machine() != "x86_64",
+    reason="The balloon device is less effective on arm64 architecture"
```

Comment:
Why? What does "less effective" mean? I think this deserves at least a brief explanation, even if `aarch64` support is experimental; maybe especially because it's experimental, to offer info for new contributors / customers in the absence of SLAs

Reply:
This is not a good reason to disable ARM tests. Why is the test not working for ARM?

Reply:
Less effective in this case means an approximately 60% drop in page throughput when it comes to inflate operations. I will, however, make the test run on arm64 and tweak the assertions to only check if there is an effect, rather than checking if there is a sufficiently large effect.